pkg/database/interface.go

```diff
@@ -0,0 +1,115 @@
+
+	ExecAll(ctx context.Context, operations *schema.ExecAllRequest) (*schema.TxHeader, error)
+
+	Count(ctx context.Context, prefix *schema.KeyPrefix) (*schema.EntryCount, error)
+	CountAll(ctx context.Context) (*schema.EntryCount, error)
+
+	ZAdd(ctx context.Context, req *schema.ZAddRequest) (*schema.TxHeader, error)
+	VerifiableZAdd(ctx context.Context, req *schema.VerifiableZAddRequest) (*schema.VerifiableTx, error)
+	ZScan(ctx context.Context, req *schema.ZScanRequest) (*schema.ZEntries, error)
+
+	// SQL-related
+	NewSQLTx(ctx context.Context, opts *sql.TxOptions) (*sql.SQLTx, error)
+
+	SQLExec(ctx context.Context, tx *sql.SQLTx, req *schema.SQLExecRequest) (ntx *sql.SQLTx, ctxs []*sql.SQLTx, err error)
+	SQLExecPrepared(ctx context.Context, tx *sql.SQLTx, stmts []sql.SQLStmt, params map[string]interface{}) (ntx *sql.SQLTx, ctxs []*sql.SQLTx, err error)
+
+	InferParameters(ctx context.Context, tx *sql.SQLTx, sql string) (map[string]sql.SQLValueType, error)
+	InferParametersPrepared(ctx context.Context, tx *sql.SQLTx, stmt sql.SQLStmt) (map[string]sql.SQLValueType, error)
+
+	SQLQuery(ctx context.Context, tx *sql.SQLTx, req *schema.SQLQueryRequest) (*schema.SQLQueryResult, error)
+	SQLQueryPrepared(ctx context.Context, tx *sql.SQLTx, stmt sql.DataSource, namedParams []*schema.NamedParam) (*schema.SQLQueryResult, error)
+	SQLQueryRowReader(ctx context.Context, tx *sql.SQLTx, stmt sql.DataSource, params map[string]interface{}) (sql.RowReader, error)
+
+	VerifiableSQLGet(ctx context.Context, req *schema.VerifiableSQLGetRequest) (*schema.VerifiableSQLEntry, error)
+
+	ListTables(ctx context.Context, tx *sql.SQLTx) (*schema.SQLQueryResult, error)
+	DescribeTable(ctx context.Context, tx *sql.SQLTx, table string) (*schema.SQLQueryResult, error)
+
+	// Transactional layer
+	WaitForTx(ctx context.Context, txID uint64, allowPrecommitted bool) error
+	WaitForIndexingUpto(ctx context.Context, txID uint64) error
+
+	TxByID(ctx context.Context, req *schema.TxRequest) (*schema.Tx, error)
+	ExportTxByID(ctx context.Context, req *schema.ExportTxRequest) (txbs []byte, mayCommitUpToTxID uint64, mayCommitUpToAlh [sha256.Size]byte, err error)
+	ReplicateTx(ctx context.Context, exportedTx []byte, skipIntegrityCheck bool, waitForIndexing bool) (*schema.TxHeader, error)
+	AllowCommitUpto(txID uint64, alh [sha256.Size]byte) error
+	DiscardPrecommittedTxsSince(txID uint64) error
+
+	VerifiableTxByID(ctx context.Context, req *schema.VerifiableTxRequest) (*schema.VerifiableTx, error)
+	TxScan(ctx context.Context, req *schema.TxScanRequest) (*schema.TxList, error)
+
+	// Maintenance
+	FlushIndex(req *schema.FlushIndexRequest) error
+	CompactIndex() error
+
+	IsClosed() bool
+	Close() error
+
+	ObjectDatabase
+}
+
+// ObjectDatabase is the interface for object database
+type ObjectDatabase interface {
+	// GetCollection returns the collection schema
+	GetCollection(ctx context.Context, req *schemav2.CollectionGetRequest) (*schemav2.CollectionInformation, error)
```

Comment:
to be changed in the proto spec, but return type should be CollectionGetResponse, as it will allow further incremental changes in the future and provide a consistent naming among all the methods

Reply:
done

Reply:
fixed